include/drivers/gpio.h

```diff
@@ -201,6 +201,13 @@ extern "C" {
  */
 #define GPIO_INT_DEBOUNCE              (1U << 19)
 
+/** Enable GPIO pin 1.8V level detection support.
+ *
+ * @note Drivers that do not support 1.8V level detection feature should ignore
+ * this flag rather than rejecting the configuration with -ENOTSUP.
+ */
+#define GPIO_INT_1P8V_SEL              (1U << 20)
```

Comment:
I don't think this is the right way to solve this.  It opens up to having more flags for 1.5V, 2.5V, 3.0V, 3.3V, 5.0V, ....

If you don't need to change the voltage configuration at runtime I'd rather see a NPCX-specific controller property in devicetree.  [This](https://github.com/torvalds/linux/blob/5c8fe583cce542aa0b84adc939ce85293de36e5e/Documentation/devicetree/bindings/arm/tegra/nvidia%2Ctegra20-pmc.yaml#L283) would be one approach: a bitmask property that indicates which pins are to be configured for which voltage.  If you can find others that have precedence in Linux they could be considered too.

If you need to change at runtime (which seems unlikely) I think you'd have to have driver-specific API extensions.

Reply:
Got it. Thanks for your suggestion. I will remove this change in gpio.h. Study adding new property now.

Reply:
Done.